data_model/1.4/clusters/DiagnosticsThread.xml

```diff
@@ -260,404 +262,404 @@ Davis, CA 95616, USA
   <attributes>
     <attribute id="0x0000" name="Channel" type="uint16">
       <access read="true" readPrivilege="view"/>
-      <quality changeOmitted="false" nullable="true" scene="false" persistence="volatile" reportable="false" sourceAttribution="false" quieterReporting="false"/>
+      <quality nullable="true"/>
       <mandatoryConform/>
     </attribute>
     <attribute id="0x0001" name="RoutingRole" type="RoutingRoleEnum">
       <access read="true" readPrivilege="view"/>
-      <quality changeOmitted="false" nullable="true" scene="false" persistence="volatile" reportable="false" sourceAttribution="false" quieterReporting="false"/>
+      <quality nullable="true"/>
       <mandatoryConform/>
     </attribute>
-    <attribute id="0x0002" name="NetworkName" type="String">
+    <attribute id="0x0002" name="NetworkName" type="string">
       <access read="true" readPrivilege="view"/>
-      <quality changeOmitted="false" nullable="true" scene="false" persistence="volatile" reportable="false" sourceAttribution="false" quieterReporting="false"/>
+      <quality nullable="true"/>
       <mandatoryConform/>
-      <constraint type="max" value="16"/>
+      <constraint type="maxLength" value="16"/>
```

Comment:
Properly a stupid Q, but in other cases this gets changed to "allowed", so why is it maxLength here?

Reply:
In the spec, this is listed as "max 16", meaning the network name could potentially be shorter. For stuff like country code, the constraint is "2", ie, it's always 2 and no more or less.